packages/api/tests/test-kinesis-consumer.js

```diff
@@ -59,92 +57,125 @@ const disabledRuleParams = Object.assign({}, commonRuleParams, {
   state: 'DISABLED'
 });
 
+/**
+ * Callback used for testing
+ *
+ * @param {*} err - error
+ * @param {Object} object - object
+ * @returns {Object} object, if no error is thrown
+ */
 function testCallback(err, object) {
   if (err) throw err;
   return object;
-};
+}
+
+test.beforeEach(async (t) => {
+  t.context.templateBucket = randomString();
+  await s3().createBucket({ Bucket: t.context.templateBucket }).promise();
+
+  t.context.stateMachineArn = randomString();
 
-test.before(async () => {
-  sinon.stub(Rule, 'buildPayload').resolves(true);
-  await manager.createTable(tableName, ruleTableParams);
+  t.context.queueUrl = await createQueue();
+
+  t.context.messageTemplate = {
+    cumulus_meta: {
+      state_machine: t.context.stateMachineArn
+    },
+    meta: { queues: { startSF: t.context.queueUrl } }
+  };
+  const messageTemplateKey = `${randomString()}/template.json`;
+  await s3().putObject({
+    Bucket: t.context.templateBucket,
+    Key: messageTemplateKey,
+    Body: JSON.stringify(t.context.messageTemplate)
+  }).promise();
+
+  sinon.stub(Rule, 'buildPayload').callsFake((item) =>
+    Promise.resolve({
+      template: `s3://${t.context.templateBucket}/${messageTemplateKey}`,
+      provider: item.provider,
+      collection: item.collection,
+      meta: get(item, 'meta', {}),
+      payload: get(item, 'payload', {})
+    })
+  );
+
+  t.context.tableName = randomString();
+  process.env.RulesTable = t.context.tableName;
+  process.env.stackName = randomString();
+  process.env.bucket = randomString();
+  process.env.kinesisConsumer = randomString();
+
+  const model = new Rule(t.context.tableName);
+  await manager.createTable(t.context.tableName, ruleTableParams);
+  await Promise.all([rule1Params, rule2Params, disabledRuleParams]
```

Comment:
```javascript
Promise.all([rule1Params, rule2Params, disabledRuleParams].map(model.create));
```

Reply:
I actually get all kinds of errors when I do that.

Reply:
oh interesting - probably that shorthand only works for promises, not iterators. Sorry for the misleading comment.